runtime/router/trie.go

```diff
@@ -95,7 +96,12 @@ func (t *Trie) Set(path string, value http.Handler) error {
 		return errors.New("path can not contain more than one *")
 	}
 
-	err := t.root.set(path, value, false, false)
+	colonAsPattern := true
+	if isWhitelisted {
+		colonAsPattern = false
+	}
```

Comment:
`colonAsPattern := !isWhitelisted` but you might as well just use:
`err := t.root.set(path, value, false, false, !isWhitelisted, isWhitelisted)`

Reply:
Having separate field for colonAsPattern would look it more easier to understand. I will go with your first approach.